### PR TITLE
docs(locators): protractor stye docs for `className`, `css`, `id`, `name`, `partialLinkText`, and `xpath`

### PR DESCRIPTION
**Pull request by cnishina**
…`className`, `css`, `id`, `name`, `partialLinkText`, and `xpath`

### CODE REVIEW

File: lib/selenium-webdriver/locators.js [hunk 1]
```diff
@@ -51,6 +73,17 @@ webdriver.By.css = webdriver.Locator.factory_('css selector');
...
  *
+ * @view
+ * <ul id="pet_id">
+ *   <li id="dog_id">Dog</li>
+ *   <li id="cat_id">Cat</li>
+ * </ul>
+ *
+ * @example
+ * // Returns the web element for dog
+ * var dog = browser.findElement(by.css('dog_id'));
```
Comment (sjelin) [inline lib/selenium-webdriver/locators.js]: Should be `by.id`

File: lib/selenium-webdriver/locators.js [hunk 2]
```diff
@@ -83,11 +116,45 @@ webdriver.By.js = function(script, var_args) {};
...
 webdriver.By.name = webdriver.Locator.factory_('name');
 
+
+/**
+ * Locates link elements whose {@linkplain webdriver.WebElement#getText visible
+ * text} contains the given substring.
+ *
+ * @view
+ * <ul>
+ *   <li><a href="https://en.wikipedia.org/wiki/Doge_(meme)">Doge meme</li>
```
Comment (sjelin) [inline lib/selenium-webdriver/locators.js]: wow

File: lib/selenium-webdriver/locators.js [hunk 3]
```diff
@@ -83,11 +116,45 @@ webdriver.By.js = function(script, var_args) {};
...
+ *   <li>Cat</li>
```
Comment (sjelin) [inline lib/selenium-webdriver/locators.js]: such view

File: lib/selenium-webdriver/locators.js [hunk 4]
```diff
@@ -83,11 +116,45 @@ webdriver.By.js = function(script, var_args) {};
...
+ * </ul>
+ *
+ * @example
+ * // Returns the 'a' web element for doge meme and navigate to that link
+ * var doge = browser.findElement(by.partialLinkText('Doge'));
```
Comment (sjelin) [inline lib/selenium-webdriver/locators.js]: so example

File: lib/selenium-webdriver/locators.js [hunk 5]
```diff
@@ -83,11 +116,45 @@ webdriver.By.js = function(script, var_args) {};
...
+ * doge.click();
```
Comment (sjelin) [inline lib/selenium-webdriver/locators.js]: wow

File: lib/selenium-webdriver/webdriver.js [hunk 6]
```diff
@@ -82,11 +82,35 @@ webdriver.WebElement.equals = function(a, b) {};
...
+ * @view
+ * <ul class="pet">
+ *   <li class="dog">Dog</a></li>
+ *   <li class="cat">Cat</li>
+ * </ul>
+ *
+ * @example
+ * // returns a promise for the web element's id and log the id
+ * var id = element(by.css('.dog')).getWebElement().getId();
+ * id.then(function(value) {
```
Comment (sjelin) [inline lib/selenium-webdriver/webdriver.js]: Maybe an `expect` here instead of a `log`?

File: lib/selenium-webdriver/locators.js (hunk 1 above)
Comment (cnishina) [inline lib/selenium-webdriver/locators.js]: Why am I bad at copying and pasting?

File: lib/selenium-webdriver/webdriver.js (hunk 6 above)
Comment (cnishina) [inline lib/selenium-webdriver/webdriver.js]: Making fix to not log.

File: lib/selenium-webdriver/webdriver.js [hunk 7]
```diff
@@ -82,11 +82,33 @@ webdriver.WebElement.equals = function(a, b) {};
...
+ * var liDog = element(by.css('.dog')).getWebElement();
+ * var liCat = liDog.getDriver().findElement(by.css('.cat'));
  */
 webdriver.WebElement.prototype.getDriver = function() {};
 
 
 /**
+ * @view
+ * <ul class="pet">
+ *   <li class="dog">Dog</a></li>
```
Comment (sjelin) [inline lib/selenium-webdriver/webdriver.js]: `</a>`?

File: lib/selenium-webdriver/webdriver.js [hunk 8]
```diff
@@ -82,11 +82,33 @@ webdriver.WebElement.equals = function(a, b) {};
...
+ *   <li class="cat">Cat</li>
+ * </ul>
+ *
+ * @example
+ * // returns the dog web element
+ * var dog = element(by.css('.dog')).getWebElement();
+ * expect(dog.getId()).not.toBe(-1);
```
Comment (sjelin) [inline lib/selenium-webdriver/webdriver.js]: Kind of a weird expectation. Is it `undefined`?  Maybe `toBe(undefined)` would be better (if that's correct)

**Comment by sjelin** [comment]
Commit message is wrong.  Otherwise LGTM

**Pull request merged**